app/api/items/update_item.go

```diff
@@ -69,18 +118,40 @@ func (srv *Service) updateItem(w http.ResponseWriter, r *http.Request) service.A
 			return err // rollback
 		}
 
-		var found bool
-		found, err = store.Permissions().MatchingUserAncestors(user).WithWriteLock().
+		var participantsGroupID *int64
+		err = store.Permissions().MatchingUserAncestors(user).WithWriteLock().
+			Joins("JOIN items ON items.id = item_id").
 			Where("item_id = ?", itemID).
 			WherePermissionIsAtLeast("edit", "all").
-			HasRows()
-		service.MustNotBeError(err)
-		if !found {
+			PluckFirst("items.contest_participants_group_id", &participantsGroupID).Error()
+
+		if gorm.IsRecordNotFoundError(err) {
 			apiError = service.ErrForbidden(errors.New("no access rights to edit the item"))
 			return apiError.Error // rollback
 		}
+		service.MustNotBeError(err)
+
+		itemData := formData.ConstructPartialMapForDB("itemWithDefaultLanguageIDAndOptionalType")
+
+		if itemData["duration"] != nil && participantsGroupID == nil {
+			service.MustNotBeError(store.RetryOnDuplicatePrimaryKeyError(func(s *database.DataStore) error {
+				participantsGroupIDTmp := s.NewID()
+				participantsGroupID = &participantsGroupIDTmp
+				service.MustNotBeError(s.Groups().InsertMap(map[string]interface{}{
+					"id": participantsGroupID, "type": "ContestParticipants",
+					"name": fmt.Sprintf("%d-participants", itemID),
```

Comment:
This part of the code is repeated and might repeat at other places as well? (when starting a contest and there is no group for it, what happens again?) 
Should be most to the store probably (the test can be minimalist)

Reply:
> when starting a contest and there is no group for it

Reply:
Fixed. There were reasons to keep it inside app.api.items:
1) The function is probably not supposed to be used outside the package.
2) Moving it into another package would require us to implement cross-package error handling.
3) Also, if moving it into another package, we would have to write many tests that would check things already tested by scenarios + tests of error handling.